packages/core/src/index.mts

```diff
@@ -108,6 +109,11 @@ export type ResolvedPromptConfig = {
   validate: (value: string) => boolean | string | Promise<string | boolean>;
 };
 
+export type SeparatorType = {
+  type?: string;
```

Comment:
I think this must be `type: 'separator'` since it cannot/shouldn't be anything else.

```suggestion
  type: 'separator';
```